pep-9999.rst

```diff
@@ -85,7 +85,112 @@ Supported Constructs
 ====================
 
 This sections lists constructs that compliant type checkers are expected
-to accept. Type stub authors can safely use these constructs.
+to accept. Type stub authors can safely use these constructs. If a
+construct is marked as "unspecified", type checkers can handle the
+construct as they best see fit or report an error. Linters should usually
+flag those constructs. Type stub authors should avoid using them.
+
+Imports
+-------
+
+Type stubs distinguish between imports that are re-exported and those
+that are only used internally. Imports are only re-exported if they
+use one of these forms:[#pep484]_
+
+* `import ... as ...`
+* `from ... import as ...`
+* `from ... import *`
+
+Stub authors can use `import foo as foo` to re-export an import using
+the same name::
+
+    from foo import func as func  # re-exported as "func"
+    from foo import Cls as OtherCls  # re-exported as "OtherCls"
+    from foo import baz  # internal use only
+    import bar as bar  # re-exported as "bar"
+    import baz  # internal use only
+
+Sub-modules are always exported when they are imported in a module.
+For example, consider the following file structure::
+
+    foo/
+        __init__.pyi
+        bar.pyi
+
+Then `foo` will export `bar` when one of the following constructs is used in
+`__init__.pyi`::
+
+    from . import bar
+    from .bar import Bar
+
+Module Level Attributes
+-----------------------
+
+Module level variables and constants can be annotated using either
+type comments or variable annotation syntax::
+
+    x: int  # recommended
+    x: int = 0
+    x = 0  # type: int
+    x = ...  # type: int
+
+The type of an unannotated variable is unspecified. Same when the annotation
```

Comment:
Sentence fragment. Maybe "The type of a variable is unspecified when the variable is unannotated or when the annotation and the assigned value disagree."